pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java

```diff
@@ -1615,4 +1615,59 @@ public void testGetTxnState() throws Exception {
         Transaction abortingTxn = transaction;
         Awaitility.await().until(() -> abortingTxn.getState() == Transaction.State.ABORTING);
     }
+
+    @Test
+    public void testUpdateSequenceIdInSyncCodeSegment() throws Exception {
```

Comment:
could you please add the test on `ClientDeduplicationTest` multi-thread send a number of messages that can send success and the consumer can receive the message.  it seems not a bug in the transaction, so don't test it with the transaction